src/daemon/dlt-daemon.c

```diff
@@ -2150,6 +2162,7 @@ void dlt_daemon_signal_handler(int sig)
         case SIGTERM:
         case SIGINT:
         case SIGQUIT:
+        case SIGSEGV:
```

Comment:
SIGSEGV needs to be handled separately. For example, after calling dlt_daemon_exit_trigger() set the SIG_DFL for SIGSEGV and raise() the same signal (or just let it hit SIGSEGV on its own again).

I really don't advise calling dlt_daemon_exit_trigger() since it calls async-unsafe functions in a signal handler, but since you already break that rule I wont complain.